File: a-m/src/main/java/m/d/a/m/p/algoprog/ru/p973.java
// https://algoprog.ru/material/p973

package m.d.a.m.p.algoprog.ru;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.BitSet;


public class p973 {
    public static void main(String[] args) throws IOException {
        BufferedReader r = new BufferedReader(new InputStreamReader(System.in));
        int k = Integer.parseInt(r.readLine());
        System.out.print(calc(k));
    }

    static int calc(int k) {
        int lim = 15485863; // 1000000th prime number
        BitSet isPrime = new BitSet(lim + 1);
        isPrime.set(0, false);
        isPrime.set(1, false);
        for (int i = 2; i <= lim; i += 1) {
            isPrime.set(i);
        }
        int c = 0;
        int nth = 0;
        for (int i = 2; i <= lim && c < k; i += 1) {
            if (isPrime.get(i)) {
                c += 1;
                nth = i;
                int j = i * i;
                while (j <= lim) {
                    isPrime.set(j, false);
                    j += i;
                }
            }
        }
        return nth;
    }
}

File: a-m/resources/ru/algoprog/p973.cpp
// https://algoprog.ru/material/p973

#include <bits/stdc++.h>

using namespace std;

int main() {
  int64_t k;
  cin >> k;
  const int64_t lim = 15485863; // 1000000th prime number
  bitset<lim+1> isPrime;
  isPrime.reset(0);
  isPrime.reset(1);
  for (int64_t i = 2; i <= lim; i += 1) {
    isPrime.set(i);
  }
  int64_t c = 0;
  int64_t nth = 0;
  for (int64_t i = 2; i <= lim && c < k; i += 1) {
    if (isPrime[i] == 1) {
      c += 1;
      nth = i;
      int64_t j = i*i;
      while (j <= lim) {
        isPrime.reset(j);
        j += i;
      }
    }
  }
  cout << nth << endl;
}